### PR TITLE
feat: implement application authentication

### PR DESCRIPTION
**Pull request by Valenteen80**

### CODE REVIEW

File: News_App/src/app/app.component.html [hunk 1]
```diff
@@ -1,13 +1,9 @@
 <body class="main">
   <app-header class="main__header"></app-header>
+  <app-authentication class="main__auth"></app-authentication>
   <app-main
     [visitorsAmount]="currentVisitorsAmount"
     class="main__body"
   ></app-main>
   <app-footer class="main__footer"></app-footer>
 </body>
-<!-- <div *ngFor="let newArticle of newArticles">
```
Comment (vadimklimchuk) [inline News_App/src/app/app.component.html:9]: Remove unused or commented lines

File: News_App/src/app/app.module.ts [hunk 2]
```diff
@@ -18,8 +20,18 @@ import { HttpClientModule } from '@angular/common/http';
...
     FilterPipe,
+    AuthenticationComponent,
+  ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    FormsModule,
+    HttpClientModule,
+    AuthModule.forRoot({
+      domain: 'news-ku.us.auth0.com',
```
Comment (vadimklimchuk) [inline News_App/src/app/app.module.ts]: You should move this credentials in environment files

File: News_App/src/app/authentication/authentication.component.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <span class="auth__status"> Вы авторизованы как {{ profile.name }}</span>
+    <div>
+      <img
+        class="auth__img"
+        [src]="profile.picture"
+        [alt]="valueAttributeImgAlt"
+      />
+    </div>
+  </div>
+  <div *ngIf="auth.isAuthenticated$ | async">
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.html]: You can't use auth service in template/ You should use for it your authetication component

File: News_App/src/app/authentication/authentication.component.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<div class="auth auth__wrapper">
+  <div class="auth" *ngIf="auth.user$ | async as user">
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.html]: the same like below comment

File: News_App/src/app/authentication/authentication.component.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        class="auth__img"
+        [src]="profile.picture"
+        [alt]="valueAttributeImgAlt"
+      />
+    </div>
+  </div>
+  <div *ngIf="auth.isAuthenticated$ | async">
+    <button class="auth-btn" (click)="logout()">Выйти</button>
+  </div>
+  <div *ngIf="(auth.isAuthenticated$ | async) === false">
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.html]: The same like above comment. And any logic should be in your compoment. You can do this compare in component

File: News_App/src/app/authentication/authentication.component.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  &__img {
+    margin-right: 20px;
+    width: 20px;
+    height: 20px;
+  }
+
+  &__status {
+    margin-right: 10px;
+    font-size: 1.2rem;
+    @include mq-small-up {
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.scss:21]: Empty line before include

File: News_App/src/app/authentication/authentication.component.scss [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  }
+
+  &-btn {
+    border: 1px solid $orange;
+    background-color: $light_beige;
+    border-radius: 25px;
+    height: 30px;
+    width: 100px;
+    &:hover {
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.scss:33]: empty line before hover

File: News_App/src/app/authentication/authentication.component.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import { AuthService } from '@auth0/auth0-angular';
+import { Profile } from '../interfaces/profile';
+
+@Component({
+  selector: 'app-authentication',
+  templateUrl: './authentication.component.html',
+  styleUrls: ['./authentication.component.scss'],
+})
+export class AuthenticationComponent implements OnInit {
+  constructor(public auth: AuthService) {}
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.ts]: What is it auth? If you use authService interface you should name property the same like "authService"

File: News_App/src/app/authentication/authentication.component.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+  public profile: Profile = {};
+  public valueAttributeImgAlt: string = 'photo';
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.ts]: It is bad name of alt attribute. Will better 'avatar, cat, dog'

File: News_App/src/app/authentication/authentication.component.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  public loginRedirect(): void {
+    this.auth.loginWithRedirect();
+  }
+
+  public logout(): void {
+    this.auth.logout();
+  }
+
+  public getProfile(): void {
+    this.auth.user$.subscribe((profile) => {
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.ts]: What type of profile?

File: News_App/src/app/main/main.component.html [hunk 11]
```diff
@@ -1,47 +1,48 @@
...
-      <div>
-        <span class="article_footer__element">{{ newArticle.author }}</span>
-        <button
-          class="article-btn article-btn__wrapper article_footer__element"
-          (click)="toggleLikeButton(newArticle)"
-          [ngClass]="{ 'article-btn--active': newArticle.isLike }"
-        >
-          Нравится
-        </button>
+<div *ngIf="auth.isAuthenticated$ | async">
```
Comment (vadimklimchuk) [inline News_App/src/app/main/main.component.html]: You can't use auth service in template

File: News_App/src/app/main/main.component.html [hunk 12]
```diff
@@ -1,47 +1,48 @@
...
+  <app-search (onAdd)="receptionEvent($event)"></app-search>
```
Comment (vadimklimchuk) [inline News_App/src/app/main/main.component.html]: It is bad name "receptionEvent"/ Will be better "searchEvent"

File: News_App/src/app/main/main.component.ts [hunk 13]
```diff
@@ -16,7 +17,10 @@ export class MainComponent implements OnInit {
   public search: string = '';
   public newArticles: Article[] = [];
 
-  constructor(public articlesService: ArticlesService) {}
+  constructor(
+    public articlesService: ArticlesService,
+    public auth: AuthService
```
Comment (vadimklimchuk) [inline News_App/src/app/main/main.component.ts]: authService instead of auth

File: News_App/src/app/main/search/search.component.ts [hunk 14]
```diff
@@ -11,7 +11,7 @@ export class FormComponent {
 
   @Output() public onAdd: EventEmitter<string> = new EventEmitter();
 
-  outputEvent() {
+  public outputEvent(): void {
```
Comment (vadimklimchuk) [inline News_App/src/app/main/search/search.component.ts]: Name "inputEvent" will better than "outputEvent"

File: News_App/src/app/services/article-servise/articles.service.ts [hunk 15]
```diff
@@ -25,7 +25,7 @@ export class ArticlesService {
       );
   }
 
-  public poseArticle(article: Article): Observable<Article> {
+  public postArticle(article: Article): Observable<Article> {
```
Comment (vadimklimchuk) [inline News_App/src/app/services/article-servise/articles.service.ts]: It is createArticle, not postArticle

File: News_App/src/app/app.component.html (hunk 1 above)
Comment (Valenteen80) [reply News_App/src/app/app.component.html:9]: deleted

File: News_App/src/app/app.module.ts (hunk 2 above)
Comment (Valenteen80) [reply News_App/src/app/app.module.ts]: rescheduled

File: News_App/src/app/authentication/authentication.component.ts (hunk 8 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.ts]: renamed

File: News_App/src/app/authentication/authentication.component.ts (hunk 9 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.ts]: renamed

File: News_App/src/app/main/main.component.html (hunk 12 above)
Comment (Valenteen80) [reply News_App/src/app/main/main.component.html]: renamed

File: News_App/src/app/main/main.component.ts (hunk 13 above)
Comment (Valenteen80) [reply News_App/src/app/main/main.component.ts]: renamed

File: News_App/src/app/main/search/search.component.ts (hunk 14 above)
Comment (Valenteen80) [reply News_App/src/app/main/search/search.component.ts]: renamed

File: News_App/src/app/services/article-servise/articles.service.ts (hunk 15 above)
Comment (Valenteen80) [reply News_App/src/app/services/article-servise/articles.service.ts]: renamed (branch name: auth)

File: News_App/src/app/app.component.html (hunk 1 above)
Comment (Valenteen80) [reply News_App/src/app/app.component.html:9]: deleted (branch name: auth)

File: News_App/src/app/authentication/authentication.component.html (hunk 3 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.html]: corrected

File: News_App/src/app/authentication/authentication.component.html (hunk 4 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.html]: corrected

File: News_App/src/app/authentication/authentication.component.html (hunk 5 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.html]: corrected

File: News_App/src/app/main/main.component.html (hunk 11 above)
Comment (Valenteen80) [reply News_App/src/app/main/main.component.html]: auth service moved from template in component.ts

File: News_App/src/app/authentication/authentication.component.ts (hunk 10 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.ts]: specified the type, redid the function and added one more

File: News_App/src/app/authentication/authentication.component.scss (hunk 7 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.scss:33]: Add empty line

File: News_App/src/app/authentication/authentication.component.scss (hunk 6 above)
Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.scss:21]: Add empty line

File: News_App/src/app/authentication/authentication.component.ts [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+import { Observable } from 'rxjs';
+import { Profile } from '../interfaces/profile';
+
+@Component({
+  selector: 'app-authentication',
+  templateUrl: './authentication.component.html',
+  styleUrls: ['./authentication.component.scss'],
+})
+export class AuthenticationComponent implements OnInit {
+  constructor(public authService: AuthService) {}
```
Comment (vadimklimchuk) [inline News_App/src/app/authentication/authentication.component.ts]: constructor should be declared after properties of class

Comment (Valenteen80) [reply News_App/src/app/authentication/authentication.component.ts]: Constructur was declared  after properties of class

**Pull request merged**